components/apimgt/org.wso2.carbon.apimgt.keymgt/src/main/java/org/wso2/carbon/apimgt/keymgt/handlers/SessionDataPublisherImpl.java

```diff
@@ -0,0 +1,348 @@
+     * @param params termination params
+     */
+    @Override
+    public void publishSessionTermination(HttpServletRequest request, AuthenticationContext context,
+            SessionContext sessionContext, Map<String, Object> params) {
+
+        OAuthConsumerAppDTO[] appDTOs = new OAuthConsumerAppDTO[0];
+        AuthenticatedUser authenticatedUser = (AuthenticatedUser) params.get("user");
+        String username = authenticatedUser.getUserName();
+        String tenantDomain = authenticatedUser.getTenantDomain();
+        String userStoreDomain = authenticatedUser.getUserStoreDomain();
+        AuthenticatedUser federatedUser;
+
+        if (authenticatedUser.isFederatedUser()) {
+            try {
+                federatedUser = buildAuthenticatedUser(authenticatedUser);
+                authenticatedUser = federatedUser;
+
+            } catch (IdentityOAuth2Exception e) {
+                log.error("Error thrown while building authenticated user in logout flow for user " + authenticatedUser
+                        .getUserName() + " : " + e);
+            }
+        }
+        try {
+            appDTOs = getAppsAuthorizedByUser(authenticatedUser);
+            if (appDTOs.length > 0) {
+                if (log.isDebugEnabled()) {
+                    log.debug("The user: " + authenticatedUser.getUserName() + " has " + appDTOs.length + " "
+                            + "OAuth apps");
+                }
+            }
+        } catch (IdentityOAuthAdminException e) {
+            log.error("Error while retrieving applications authorized for the user " + authenticatedUser.getUserName()
+                    + " : " + e);
+        }
+
+        for (OAuthConsumerAppDTO appDTO : appDTOs) {
+            if (StringUtils.equalsIgnoreCase("admin_store",appDTO.getApplicationName()) ||
+                    StringUtils.equalsIgnoreCase("admin_publisher",appDTO.getApplicationName())) {
+                Set<AccessTokenDO> accessTokenDOs = null;
+                try {
+                    // Retrieve all ACTIVE or EXPIRED access tokens for particular client authorized by this user
+                    accessTokenDOs = OAuthTokenPersistenceFactory.getInstance().getAccessTokenDAO()
+                            .getAccessTokens(appDTO.getOauthConsumerKey(), authenticatedUser,
+                                    authenticatedUser.getUserStoreDomain(), true);
+                } catch (IdentityOAuth2Exception e) {
+                    log.error("Error while retrieving access tokens for the application " + appDTO.getApplicationName()
+                            + "and the for user " + authenticatedUser.getUserName() + " : " + e);
+                }
+                AuthenticatedUser authzUser;
+                if (accessTokenDOs != null) {
+                    Set<String> accessTokens = new HashSet<>();
+                    for (AccessTokenDO accessTokenDO : accessTokenDOs) {
+                        accessTokens.add(accessTokenDO.getAccessToken());
```

Comment:
Do we use this accessToken set anywhere?

Reply:
nope, we are just setting the iterated access tokens to this accessTokens Set, it doesn't have any other use case yet.

Reply:
Shall we remove it, if there is no use of it?

Reply:
removed and resolved